File: .github/workflows/StackOverflow.py
from stackapi import StackAPI
import time
import requests
import sys

webHookURL = sys.argv[1]

SITE = StackAPI('stackoverflow')

currenttime = int(time.time()) #Get current time in Unix Epcot Time

thirtyMinsAgo = currenttime - 1800 #rewind 30 minutes since that's the frequency the job run's at. 

posts = SITE.fetch('questions', fromdate = thirtyMinsAgo, todate = currenttime, tagged = 'Julia')

headers = {
    'Content-type': 'application/json',
}


for k,v in posts.items():
    if k == "items":
        for item in v:
            title = ""
            owner = ""
            answer_count = ""
            link = ""            
            for key,value in item.items():
                #print("Key: " + str(key) + " Value: " + str(value)) #Used for DEBUG
                
                if key == "title":
                    title = value
                    #print("Title1: " + str(title)) #Used for DEBUG                    
                elif key == "owner":
                    for name_key, name_value in value.items():
                        if name_key == "display_name":
                            owner = name_value
                elif key == "answer_count":
                    answer_count = value
                elif key == "link":
                    link = value
                   
                #End of if
                
            #End of for
            
            currenttime = int(time.time()) 
            data = ('{ "attachments": [ { "color": "#36a64f", "title": "%s", "title_link": "%s", "text": "%s", "footer": "Slack API", "ts": %d, "fields": [ { "title": "Answer Count: %s", "short": false } ] } ] }' % (str(title), link, link, currenttime, str(answer_count)))
            
            # print(data + "\n") #Debug. 
            
            # WARNING: Un-commenting this line will push to prod. 
            response = requests.post(webHookURL, headers=headers, data=data)
            print(response)
            # WARNING     
                            
        #End of internal for loop. 
    #End of if. 
#End of for

File: StackOverflow.jl
#=
Main:
- Julia version: 1.0.4
- Author: pepsipu
- Date: 2019-12-04
=#

using StackOverflow
using HTTP
using Dates
using JSON

endpoint = "/services/TR8BQV1U4/BR2K9KQC9/x3JoBxwBBHA5wGuld5pXtfya"
epoch = trunc(Int, Dates.datetime2unix(Dates.now()))
thirtyMinutesAgo = epoch - 3600
questions = getrecentquestionsfortag(tag = "julia", fromdate = string(thirtyMinutesAgo), todate = string(epoch))
for question in questions
    data = Dict(
        "attachments" => [
            "color" => "#36a64f",
            "title" => question.title[1],
            "title_link" => question.link[1],
            "text" => question.link[1],
            "footer" => "Slack API",
            "ts" => string(trunc(Int, Dates.datetime2unix(Dates.now()))),
            "fields" => [
                Dict(
                    "title" => "Answer Count: $(question.answer_count[1])",
                    "short" => false
                )
            ]
        ]
    )
    HTTP.setuseragent!("Julia Slack Bot")
    r = HTTP.post(
        "https://hooks.slack.com" * endpoint,
        ["Content-Type" => "application/json"],
        JSON.json(data)
    )
end